tips/TIP-0040/tip-0040.md

```diff
@@ -0,0 +1,848 @@
+protocol-defined duration in seconds. A new validator committee is selected for each epoch. Shortly before an epoch
+ends, the current activity of all registered validators, as well as the pools' stakes, are determined. Then, a committee
+is selected out of all the registered stakers with enough block issuance activity, depending on their pool stake. Once
+selected, each validator is responsible for issuing _Validation Blocks_ across the epoch through which they vote on
+conflicts. A performance factor is calculated based on how many such blocks they issued for each slot and how many they
+were expected to issue. This factor determines the amount of _Mana rewards_ the validator's pool receives.
+
+After an epoch ends, the delegators of the pool can claim their rewards. The validator may continue to stake or end its
+stake by going through an unbonding period of its locked tokens. After this period ends, the validator can unlock their
+IOTA coins and claim their Mana rewards.
+
+## Validator Registration
+
+Accounts are considered registered as committee candidates for epoch `Candidate Epoch` if they satisfy all of the
+following conditions:
+
+- The account has a [_Staking Feature_](../TIP-0042/tip-0042.md#staking-feature) in their _Features_ where
+  `End Epoch >= Candidate Epoch`.
+- The account issues a Basic Block in a slot `Block Slot` with a _Candidacy Announcement Payload_ where
+  `Start Slot(Candidate Epoch - 1) <= Block Slot <= Registration Slot(Candidate Epoch)`. `Epoch Start Slot` and
+  `Registration Slot` are defined in [committee selection](#committee-selection).
+
+### Candidacy Announcement
+
+A _Candidacy Announcement Payload_ does not contain any data as it is only used to signal candidacy. It is serialized as
+follows:
+
+<details>
+  <summary>Candidacy Announcement</summary>
+  <blockquote>Signals candidacy for committee selection for the epoch after the one in which it is issued.</blockquote>
+</details>
+<table>
+  <tr>
+    <td>
+      <b>Name</b>
+    </td>
+    <td>
+      <b>Type</b>
+    </td>
+    <td>
+      <b>Description</b>
+    </td>
+  </tr>
+  <tr>
+    <td>Payload Type</td>
+    <td>uint8</td>
+    <td>Set to <strong>value 2</strong> to denote a <i>Candidacy Announcement Payload</i>.</td>
+  </tr>
+</table>
+
+## Committee Selection
+
+![](./assets/selection-timing.png)
+
+_This figure gives an overview of the timing around committee selection in epochs. Note that the exact length of those
+windows is not accurately depicted here and instead depends on the value of the corresponding Protocol Parameters._
+
+The following explains how the committee selection for a certain epoch `n` works, with the following time boundaries
+used in the algorithm:
+
+- Let `Epoch Start Slot(n)` be the first slot index of epoch `n`.
+- Let `Registration Slot(n)` be the registration slot with index `Epoch Start Slot(n) - Epoch Nearing Threshold - 1`. To
```

Comment:
Is epoch start slot correct? Here it's the end slot
https://github.com/iotaledger/iota.go/blob/42c1479d3f301ce7ca44871edebd4283727503ac/vm/nova/vm.go#L1042